app/controllers/tournaments_controller.rb

```diff
@@ -1,16 +1,36 @@
+require 'json'
+
 class TournamentsController < ApplicationController
 
+  include TournamentsHelper
   skip_before_action :verify_authenticity_token
 
   def index
   end
 
   def create
-    Tournament.create!(company_name: params[:company_name], start_date: params[:start_date], end_date: params[:end_date])
+    begin
+      new_tournament = Tournament.create!(company_name: params[:company_name], start_date: params[:start_date], end_date: params[:end_date])
+      results = params[:results]
+      results.each do |result|
+        new_user = User.create!(first_name: result[:firstName], last_name: result[:lastName], email: result[:email])
+        temp = Result.create!(user_id: new_user.id, tournament_id: new_tournament.id , answers: result[:answers])
+      end
+    rescue 
+      render json: "message: error occurred", status: :bad_request
```

Comment:
It's nicer to rescue errors in the ApplicationController
This way you don't have to wrap each endpoint with a begin/rescue, it's done automatically for all of the endpoints
You can just add something like this there:

```ruby
rescue_from StandardError do |exception|
      render json: { error: "Internal Server Error" }, status: :internal_server_error
end
```

Reply:
However I'm not sure here, since you're validating input and returning bad_request by creating the objects and relying on the model validation, so maybe it is ok to just leave it
Another choice is to use `create` instead of `create!`, check for errors and return bad_request when there are errors
Also, it's more :unprocessable_entity than it is :bas_request - https://stackoverflow.com/questions/16133923/400-vs-422-response-to-post-of-data#:~:text=The%20422%20(Unprocessable%20Entity)%20status,was%20unable%20to%20process%20the